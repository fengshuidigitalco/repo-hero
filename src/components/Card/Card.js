import React from 'react';
import {
  Avatar,
  Card as MUICard,
  CardContent,
  CardHeader,
  createStyles,
  makeStyles,
} from '@material-ui/core';
import { blue } from '@material-ui/core/colors';
import cx from 'classnames';
import { node, number, object, string } from 'prop-types';

const useStyles = makeStyles((theme) =>
  createStyles({
    avatar: {
      backgroundColor: (props) => blue[props.colorNumber],
    },
    footer: {
      ...theme.flexContainer.row,
      justifyContent: 'space-around',
    },
    cardHeader: {
      fontSize: '20px',
      fontWeight: 700,
    },
    content: {
      height: 110,
      overflow: 'hidden',
      marginBottom: '40px',
      textOverflow: 'ellipsis',
    },
    root: {
      boxShadow:
        '0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%)',
      maxWidth: '90vw',
      marginBottom: '1em',
      minWidth: 275,
      paddingBottom: '1em',
      width: '100%',
      [theme.breakpoints.up('md')]: {
        margin: '1em',
      },
      [theme.breakpoints.up('lg')]: {
        maxWidth: 300,
      },
    },
  }),
);

Card.propTypes = {
  id: string.isRequired,
  classOverrides: object,
  title: string.isRequired,
  content: node.isRequired,
  footer: node,
  color: number.isRequired,
  subHeader: string,
  contentClassName: string,
};
export function Card({
  id,
  classOverrides,
  title,
  content,
  footer,
  color,
  subHeader,
  contentClassName,
  ...rest
}) {
  const props = { colorNumber: color };
  const classes = useStyles(props);
  const cardId = `${id}-card`;

  return (
    <MUICard
      id={cardId}
      className={classes.root}
      classes={classOverrides}
      variant="outlined"
      {...rest}>
      <CardHeader
        avatar={
          <Avatar aria-label="title" className={classes.avatar}>
            {title.charAt(0)}
          </Avatar>
        }
        title={title}
        subheader={subHeader}
        classes={{ title: classes.cardHeader }}
      />
      <CardContent className={cx(classes.content, contentClassName)}>
        {content}
      </CardContent>
      {footer && (
        <div className={classes.footer} id={`${cardId}-footer`}>
          {footer}
        </div>
      )}
    </MUICard>
  );
}
