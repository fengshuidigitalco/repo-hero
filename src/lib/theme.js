import { createMuiTheme } from '@material-ui/core/styles';
import { blue, green, purple } from '@material-ui/core/colors';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: purple[500],
      dark: '#259857',
      light: '#62c48d',
      contrastText: '#ffffff',
    },
    secondary: {
      main: green[700],
      dark: '#FF902C',
      light: '#F0BC61',
      contrastText: '#ffffff',
    },
    warning: {
      main: blue[700],
    },
  },
  flexContainer: {
    row: {
      display: 'flex',
      flexDirection: 'row',
    },
    column: {
      display: 'flex',
      flexDirection: 'column',
    },
  },
  breakpoints: {
    values: {
      xs: 300,
      mobile: 415,
      sm: 500,
      smTab: 610,
      md: 650,
      lg: 800,
      xl: 1025,
    },
  },
});
