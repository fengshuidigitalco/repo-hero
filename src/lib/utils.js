import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core';
import { theme } from './theme';

export const renderWithTheme = (children: JSX.Element) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
