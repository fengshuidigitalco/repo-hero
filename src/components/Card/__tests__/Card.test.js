import { Card } from '../Card';
import { renderWithTheme } from '../../../lib';
const PROPS = {
  id: 'test',
  title: 'Test Card',
  content: 'Card Content',
  color: 300,
};

const testId = `${PROPS.id}-card`;

describe('card', () => {
  it('renders', () => {
    renderWithTheme(<Card {...PROPS} />);
    const renderedCard = document.getElementById(testId);
    expect(renderedCard).toMatchSnapshot();
  });

  it('renders with footer', () => {
    renderWithTheme(<Card {...PROPS} footer="Card footer" />);
    const cardFooter = document.getElementById(`${testId}-footer`);
    expect(cardFooter).toBeInTheDocument();
  });
});
