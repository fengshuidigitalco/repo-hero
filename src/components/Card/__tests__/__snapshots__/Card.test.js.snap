// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-outlined MuiPaper-rounded"
  id="test-card"
>
  <div
    class="MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-avatar"
    >
      <div
        aria-label="title"
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-1 makeStyles-avatar-6 MuiAvatar-colorDefault"
      >
        T
      </div>
    </div>
    <div
      class="MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeader-3 MuiTypography-body2 MuiTypography-displayBlock"
      >
        Test Card
      </span>
    </div>
  </div>
  <div
    class="MuiCardContent-root makeStyles-content-4"
  >
    Card Content
  </div>
</div>
`;
